data-processing/entities/terms/extractor.py

```diff
@@ -10,110 +9,169 @@
-            for i in matches_tuples:
-                matches.add(tuple(i))
-
-            # Getting the start and end indices of the matched terms
-            for match in matches:
-                start_index_of_string = text_segment_string.find(match[0])
-                end_index_of_string = start_index_of_string + len(match[0]) - 1
-                matched_terms.append(match[0])
-                matched_terms_context_str.append(match[1])
-                matched_terms_start.append(
-                    text_segment.tex_start + start_index_of_string)
-                matched_terms_end.append(
-                    text_segment.tex_start + end_index_of_string)
-
-        data_frame_dict = {"term": matched_terms, "start": matched_terms_start,
-                           "end": matched_terms_end, "context_str": matched_terms_context_str}
-        data_frame = pd.DataFrame(data_frame_dict).sort_values(by=['end'])
-
-        # Getting rid of cases where both "Neural Network" and "Neural Networks"
-        # are included (when they start at the same index, i.e. same word except with an extra "s")
-        data_frame = data_frame.drop_duplicates(subset=['start'], keep='last')
-        data_dict = data_frame.to_dict('records')
-        sorted_string = json.dumps(data_dict, indent=4, sort_keys=False)
-        json_data = json.loads(sorted_string)
-
-        index = 0
-        for term_object in json_data:
-            yield Term(
-                name=term_object["term"],
-                # start position of the term in the TeX
-                start=term_object["start"],
-                end=term_object["end"],  # end position
-                id_=str(index),  # unique ID for each term (index)
-                # tex_path (what was passed into the function)
-                tex_path=tex_path,
-                tex=tex,  # includes all of the spaces and the LaTeX junk
-                # context_tex=term_object["context_str"],  # tex + 20 characters before + 20 characters after
-                # Assuming DEFAULT_CONTEXT_SIZE is 20 characters
-                context_tex=tex[start - \
-                                DEFAULT_CONTEXT_SIZE: end + DEFAULT_CONTEXT_SIZE]
-            )
-            index += 1
+	"""
+	Extract plaintext terms from TeX, with the locations of the start and end characters they correspond to in
+	the input TeX strings.
+	"""
+
+	# Extracting all the terms, definition_text and definition_html from glossary
+	terms_and_def_text_html = []
+
+	for obj in glossary:
+		new_triple = []
+
+		#Making terms lower case in one step
+		new_triple.append(obj["terms"].lower())
+		new_triple.append(obj["definition_text"])
+		new_triple.append(obj["definition_html"])
+		terms_and_def_text_html.append(new_triple)
+
+	matched_terms = []
+	matched_definition_texts = []
+	matched_definition_htmls = []
+	matched_terms_start = []
+	matched_terms_end = []
+	matched_terms_context_str = []
```

Comment:
They are inside the parse method now; i realized I need to make a new .__init__ function because we have to add a new parameter for the TermExtractor (namely, a parameter for the input glossary).